packages/backend-for-frontend/src/routers/catalogRouter.ts

```diff
@@ -968,18 +968,43 @@ const catalogRouter = (
       }
     })
     .post(
-      "/templates/eservices/:eServiceId/descriptors/:descriptorId/interface",
+      "/templates/eservices/:eServiceId/descriptors/:descriptorId/interface/soap",
       async (req, res) => {
         const ctx = fromBffAppContext(req.ctx, req.headers);
         try {
-          const response =
-            await eserviceTemplateService.addEserviceInterfaceByTemplate(
+          const eserviceId =
+            await eserviceTemplateService.addEserviceInterfaceSoapByTemplate(
               unsafeBrandId(req.params.eServiceId),
               unsafeBrandId(req.params.descriptorId),
               req.body,
               ctx
             );
-          return res.status(200).send(bffApi.CreatedResource.parse(response));
+          return res.status(200).send(bffApi.CreatedResource.parse(eserviceId));
+        } catch (error) {
+          const errorRes = makeApiProblem(
+            error,
+            addEServiceInterfceByTemplateErrorMapper,
+            ctx.logger,
+            ctx.correlationId,
+            `Error adding interface for eService ${req.params.eServiceId}`
+          );
+          return res.status(errorRes.status).send(errorRes);
+        }
+      }
+    )
+    .post(
+      "/templates/eservices/:eServiceId/descriptors/:descriptorId/interface/rest",
+      async (req, res) => {
+        const ctx = fromBffAppContext(req.ctx, req.headers);
+        try {
+          const eserviceId =
+            await eserviceTemplateService.addEserviceInterfaceRestByTemplate(
```

Comment:
Same
```suggestion
            await eserviceTemplateService.addEServiceTemplateInstanceInterfaceRest(
```

Reply:
this commit [d1e9dde](https://github.com/pagopa/interop-be-monorepo/pull/1643/commits/d1e9dde2510d1e7615a163496e73612e3c097faa) contains this refactor